src/main/java/frc/robot/Subsystems/Drivetrain.java

```diff
@@ -0,0 +1,190 @@
+ package frc.robot.Subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj.AnalogGyro;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.DriveStyle;
+import frc.robot.Constants.JoystickScaling;
+
+public class Drivetrain extends SubsystemBase {
+    //initialize motor controllers, encoders, gyro, and differential drive odometry
+  private CANSparkMax m_leftMotor_front, m_leftMotor_back, m_rightMotor_front, m_rightMotor_back;
+  private MotorControllerGroup leftMotors;
+  private MotorControllerGroup rightMotors;
+  private RelativeEncoder leftEncoder, rightEncoder;
+  private AnalogGyro gyro;
+
+  private DifferentialDriveOdometry driveOdometry;
+
+  private static double scaleFactorA = 0.4;
+  private static double scaleFactorB = 0.55;
+  private static double scaleFactorC = 0.35;
+  private static double scaleFactorD = 0.6;
+  private double leftSpeed;
+  private double rightSpeed;
+
+    public Drivetrain() {
+        m_leftMotor_front = new CANSparkMax(Constants.MOTOR_LEFT_MASTER_ID, MotorType.kBrushless);
+        m_leftMotor_back = new CANSparkMax(Constants.MOTOR_LEFT_SLAVE_ID, MotorType.kBrushless);
+        m_rightMotor_front = new CANSparkMax(Constants.MOTOR_RIGHT_MASTER_ID, MotorType.kBrushless);
+        m_rightMotor_back = new CANSparkMax(Constants.MOTOR_RIGHT_SLAVE_ID, MotorType.kBrushless);
+
+        leftMotors = new MotorControllerGroup(m_leftMotor_front, m_leftMotor_back);
+        rightMotors = new MotorControllerGroup(m_rightMotor_front,m_rightMotor_back);
+        rightMotors.setInverted(true);
+
+        leftEncoder = m_leftMotor_front.getEncoder();
+        rightEncoder = m_rightMotor_front.getEncoder();
+
+        gyro = new AnalogGyro(Constants.ANALOG_GYRO_PORT);
+        
+        driveOdometry = new DifferentialDriveOdometry(gyro.getRotation2d(), 0, 0);
+        resetEncoders();
+    }
+
+    //converts turn and forward speeds to left and right motor speeds
+    public void drive(double turn, double forward, JoystickScaling scaling, DriveStyle driveStyle) {
+        double turnPower = scaleInputs(turn, scaling);
+        double forwardPower = scaleInputs(forward, scaling);
+
+        switch(driveStyle) {
```

Comment:
What's the purpose of having these different drive styles?

Reply:
There's no major difference, but different drive styles get the chassis driving differently (arcade tank is at least theoretically better at conserving the robot's momentum than normal arcade, for example)